### ISSUE TITLE
The JSON value could not be converted - ObservableCollection<T>

### ISSUE THREAD

**Issue by Symbai** [dotnet/runtime#389]
I cannot deserialize the following JSON to an ObservableCollection<T> (required for WPF applications) but successfully deserialize it to a List<T>. In such a case deserializing to an ObservableCollection should be possible as well?!

#### Exception:
```{"The JSON value could not be converted to System.Collections.Generic.ICollection`1[WpfApp12.Testing+Foo]. Path: $.MyList | LineNumber: 1 | BytePositionInLine: 22."}```

#### JSON:
```
{
	"MyList": [
		{
			"Name": "Paul"
		},
		{
			"Name": "Mike"
		}
	]
}
```
#### Fails:
```
public class Foo
{
    public ObservableCollection<FooMember> MyList { get; set; }
}
public class FooMember
{
    public string Name { get; set; }
}
```
#### Succeeds:
```
public class Foo
{
    public List<FooMember> MyList { get; set; }
}
public class FooMember
{
    public string Name { get; set; }
}
```

**Comment by layomia** [comment]
cc @maryamariyan on bot labeling.

**Comment by layomia** [comment]
Support for `ObservableCollection<T>` was added for master/5.0 in https://github.com/dotnet/corefx/pull/40654 and 3.1 in https://github.com/dotnet/corefx/pull/41523. Can you please try one of these releases?

- https://github.com/dotnet/core-sdk#installers-and-binaries 
- https://github.com/dotnet/core/blob/master/release-notes/3.1/preview/3.1.0-preview3.md

**Comment by Symbai** [comment]
@layomia Thanks for answering. I can confirm it is fixed in 5.0 alpha
